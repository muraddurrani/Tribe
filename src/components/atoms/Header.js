import React from 'react'
import { View, ImageBackground, StyleSheet } from 'react-native'
import { Text } from 'react-native-elements'
import colours from '../../styles/colours'

function Header(props) {
  return (
    <View style = {{...styles.container, ...props.style}} >
      <ImageBackground source = {require('../../assets/images/Header_Background.png')} style = {styles.background}>
      <Text h2Style = {styles.header} h2>{props.children}</Text>
    </ImageBackground>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colours.primary
  },
  background: {
    width: '100%',
    height: 120,
    justifyContent: 'flex-end'
  },
  header: {
    color: colours.gray0,
    marginLeft: 20,
    marginBottom: 20
  }
})

export default Header
