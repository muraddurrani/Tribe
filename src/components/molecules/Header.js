import React from 'react'
import { StyleSheet } from 'react-native'
import { Icon, Text } from 'react-native-elements'
import LinearGradient from 'react-native-linear-gradient'
import colours from '../../styles/colours'

function Header({title, onPress, icon = false}) {
  return (
    <LinearGradient start={{x: 0, y: 0}} end={{x: 1, y: 0}} colors = {[colours.primary, colours.secondary]} style = {styles.container}>
      {icon && <Icon name = "chevron-left" color = {colours.gray0} size = {30} containerStyle = {styles.icon} onPress = {onPress} />}
      <Text h2Style = {styles.title} h2>{title}</Text>
    </LinearGradient>
  )
}

const styles = StyleSheet.create({
  container: {
    height: 70,
    width: '100%',
    paddingVertical: 15,
    paddingHorizontal: 20,
    alignItems: 'flex-end',
    flexDirection: 'row'
  },
  title: {
    color: colours.gray0
  },
  icon: {
    marginRight: 20
  }
})

export default Header
