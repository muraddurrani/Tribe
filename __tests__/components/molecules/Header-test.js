import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../../src/components/molecules/Header'

jest.useFakeTimers()

it('Header renders correctly', () => {
  const tree = renderer
    .create(<Header title = {'Header'} onPress = {console.log('Header Pressed') }/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});