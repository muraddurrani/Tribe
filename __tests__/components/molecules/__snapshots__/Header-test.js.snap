// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294924902,
      4294942810,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 70,
      "paddingHorizontal": 20,
      "paddingVertical": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fdfdfd",
        "fontSize": 36,
      }
    }
  >
    Header
  </Text>
</BVLinearGradient>
`;
